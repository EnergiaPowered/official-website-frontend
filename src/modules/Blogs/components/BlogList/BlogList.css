.blogs-container {
  margin: 0 auto;
  padding: 0;
}

.blogcard {
  align-items: stretch;
  min-height: 400px;
  margin: 20px 0;
}

.blogcard a {
  text-decoration: none;
}

.blogcard__content {
  background-color: #ffffffab;
  cursor: pointer;
  height: 100%;
  transition: all 200ms ease-in;
}

.blogcard__content:hover {
  transform: scale(1.02);
}

.blogcard__content .img-container {
  width: 100%;
  height: 65%;
  background-color: #00000057;
}

.blogcard__content .img-container img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.blogcard__body {
  color: #010e30;
  padding: 5%;
}

.blogcard__heading {
  color: #010e30;
  font-weight: bold;
  font-size: 28px;
  text-transform: capitalize;
}