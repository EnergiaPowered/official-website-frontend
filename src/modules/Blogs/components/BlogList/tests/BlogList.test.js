import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import { renderer, create } from "react-test-renderer";
import BlogList from "./../BlogList";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BlogList />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot 1", () => {
    const bloglist = create(<BlogList />).toJSON();
    expect(bloglist).toMatchSnapshot();
});
/*
it("renders correctly with props", () => {
    const blogList = <BlogList setIsBlogOpened="true" setClickedBlog="" />;
    const { getByTestId } = render(blogList);
    expect(getByTestId("bloglist")).toHaveTextContent("true", "");
});
it("matches snapshot 2", () => {
    const bloglist = create(
        <BlogList setIsBlogOpened="true" setClickedBlog="" />
    ).toJSON;
    expect(bloglist).toMatchSnapshot();
});
//for the date
/*
const date = "21 JAN 2015";
const format = "DD MMM YYYY";
it("render valueToDate utility with empty value", () => {
    expect(valueToDate("", format)).toEqual(null);
});
it("check value is instanceof moment", () => {
    expect(valueToDate(date, format) instanceof moment).toBeTruthy();
});
*/
