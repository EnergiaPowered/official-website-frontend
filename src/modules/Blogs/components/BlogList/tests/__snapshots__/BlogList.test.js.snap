// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="container blogs-container row"
  data-testid="bloglist"
>
  <div
    className="col-12 d-flex flex-column align-items-center p-5"
  >
    <span
      className="fa fa-spinner fa-pulse fa-3x fa-fw text-primary"
      style={
        Object {
          "fontSize": "24px",
        }
      }
    />
    <p
      className="pt-3"
      style={
        Object {
          "fontSize": "24px",
        }
      }
    >
      Loading...
    </p>
  </div>
</div>
`;
